pkg/monitoring/monitoring.go

```diff
@@ -72,15 +72,21 @@ func OwnedAppListOptions(clusterID, appName, appTargetNamespace string) metav1.L
 	}
 }
 
-func CopyCreatorID(toAnnotations, fromAnnotations map[string]string) map[string]string {
-	if val, exist := fromAnnotations[cattleCreatorIDAnnotationKey]; exist {
-		if toAnnotations == nil {
-			toAnnotations = make(map[string]string, 2)
-		}
+func CopyCreatorIDAndOverwroteAppAnswers(toAnnotations, fromAnnotations map[string]string) map[string]string {
+	if toAnnotations == nil {
+		toAnnotations = make(map[string]string, 2)
+	}
 
+	// copy creatorID
+	if val := fromAnnotations[cattleCreatorIDAnnotationKey]; val != "" {
```

Comment:
I think we should check for key existence like `val, ok :=  fromAnnotations[cattleCreatorIDAnnotationKey]`
If there is an empty value for the annotation, it should be copied too.

Reply:
updated